clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/rocksdb/RocksDBStoragePartition.java

```diff
@@ -484,16 +484,12 @@ public synchronized void endBatchWrite() {
      * the last SST file written is finished.
      */
     rocksDBSstFileWriter.ingestSSTFiles(rocksDB, columnFamilyHandleList);
-
-    if (blobTransferEnabled) {
-      createSnapshot();
-    }
   }
 
   @Override
   public synchronized void createSnapshot() {
     if (blobTransferEnabled) {
-      rocksDBSstFileWriter.createSnapshot(rocksDB);
+      BlobSnapshotManager.createSnapshotForBatch(rocksDB, fullPathForTempSnapshotFileDir);
```

Comment:
I don't recall it needed `fullPathForTempSnapshotFileDir` and `fullPathForPartitionDBSnapshot`. Looks like they are thee same so we are eliminating one of them, correct?

Reply:
We have some duplicated logics here, not your changes but from previous changes. We may address them too.

In `RocksDBStoragePartition` constructor, we first check whether `blobTransferEnabled` is enabled or not to determine whether we need to create `fullPathForTempSnapshotFileDir`. In this method, we check this flag again to determine whether we need to run `BlobSnapshotManager.createSnapshotForBatch`. In the `createSnapshotForBatch` method, we check whether `fullPathForTempSnapshotFileDir` is present or not which is actually already guarded by the logic in `RocksDBStoragePartition` constructor. I feel we could remove one of these duplicated checks.

Reply:
Yes, lets only use fullPathForPartitionDBSnapshot. 
It is a good point! lets remove the path check at createSnapshotForBatch.